www/js/onboarding/onboardingHelper.ts

```diff
@@ -58,15 +62,22 @@ export function getPendingOnboardingState(): Promise<OnboardingState> {
 };
 
 async function readConsented() {
-  const StartPrefs = getAngularService('StartPrefs');
-  return StartPrefs.readConsentState().then(StartPrefs.isConsented) as Promise<boolean>;
+  return readConsentState().then(isConsented) as Promise<boolean>;
 }
 
-async function readIntroDone() {
+export async function readIntroDone() {
   return storageGet(INTRO_DONE_KEY).then((read_val) => !!read_val) as Promise<boolean>;
 }
 
 export async function markIntroDone() {
   const currDateTime = DateTime.now().toISO();
-  return storageSet(INTRO_DONE_KEY, currDateTime);
+  return storageSet(INTRO_DONE_KEY, currDateTime)
+    .then(() => {
+      //handle "on intro" events
+      logDebug("intro done, calling registerPush and storeDeviceSettings");
+      const PushNotify = getAngularService("PushNotify");
+      const StoreSeviceSettings = getAngularService("StoreDeviceSettings");
```

Comment:
Is this the recommended way to handle this in react? The reason that I had an event-driven implementation before this was to simplify the dependency chain. With this, we are adding dependencies between startPrefs and push notify and store device settings.